datumaro/plugins/data_formats/video.py

```diff
@@ -124,3 +127,101 @@ def __iter__(self):
     def get(self, id, subset=None):
         assert subset == self._subset, "%s != %s" % (subset, self._subset)
         return super().get(id, subset or self._subset)
+
+
+class VideoKeyframesImporter(VideoFramesImporter):
+    """
+    Reads video frames as a dataset.
+    """
+
+    @classmethod
+    def build_cmdline_parser(cls, **kwargs):
+        parser = super().build_cmdline_parser(**kwargs)
+        parser.add_argument(
+            "-t",
+            "--threshold",
+            type=float,
+            default=0.3,
+            help="Similarity threshold (default: %(default)s)",
+        )
+
+        parser.add_argument(
+            "-resize",
+            "--resize",
+            type=float,
+            default=(64, 64),
+            help="Image size for comuting ZNCC score (default: %(default)s)",
+        )
+        return parser
+
+    @classmethod
+    def find_sources(cls, path):
+        if not osp.isfile(path):
+            return []
+        return [{"url": path, "format": VideoKeyframesBase.NAME}]
+
+
+class VideoKeyframesBase(VideoFramesBase):
+    def __init__(
+        self,
+        url: str,
+        *,
+        subset: Optional[str] = None,
+        name_pattern: str = "%06d",
+        step: int = 1,
+        start_frame: int = 0,
+        end_frame: Optional[int] = None,
+        resize: Tuple[int, int] = (64, 64),
+        threshold: float = 0.3,
+    ) -> None:
+        self._subset = subset or DEFAULT_SUBSET_NAME
+        super().__init__(
+            url=url,
+            step=step,
+            start_frame=start_frame,
+            end_frame=end_frame,
+            name_pattern=name_pattern,
+        )
+
+        assert osp.isfile(url), url
```

Comment:
```suggestion
        super().__init__(
            url=url,
            subset=subset,
            step=step,
            start_frame=start_frame,
            end_frame=end_frame,
            name_pattern=name_pattern,
        )
```

Some lines are redundant in `VideoFramesBase.__init__()`

Reply:
thank you, I corrected.